SwiftGit2Tests/RepositorySpec.swift

```diff
@@ -645,6 +645,214 @@ class RepositorySpec: QuickSpec {
+				expect(repo.checkout(branch, strategy: CheckoutStrategy.None).error).to(beNil())
+
+				let status = repo.status()
+
+				expect(status.value?.count).to(equal(expectedCount))
+			}
+
+			it("Should accurately report status for repositories with status") {
+				let expectedCount = 5
+				let expectedNewFilePaths = [
+					"stage-file-1",
+					"stage-file-2",
+					"stage-file-3",
+					"stage-file-4",
+					"stage-file-5",
+				]
+				let expectedOldFilePaths = [
+					"stage-file-1",
+					"stage-file-2",
+					"stage-file-3",
+					"stage-file-4",
+					"stage-file-5",
+				]
+
+				let repoWithStatus = Fixtures.sharedInstance.repository(named: "repository-with-status")
+				let branchWithStatus = repoWithStatus.localBranch(named: "master").value!
+				expect(repoWithStatus.checkout(branchWithStatus, strategy: CheckoutStrategy.None).error).to(beNil())
+
+				let statuses = repoWithStatus.status().value!
+
+				var newFilePaths: [String] = []
+				for status in statuses {
+					newFilePaths.append((status.headToIndex?.newFile?.path)!)
+				}
+				var oldFilePaths: [String] = []
+				for status in statuses {
+					oldFilePaths.append((status.headToIndex?.oldFile?.path)!)
+				}
+
+				expect(statuses.count).to(equal(expectedCount))
+				expect(newFilePaths).to(equal(expectedNewFilePaths))
+				expect(oldFilePaths).to(equal(expectedOldFilePaths))
+			}
+		}
+
+		describe("Repository.diff") {
+			it("Should have accurate delta information") {
+				let expectedCount = 13
+				let expectedNewFilePaths = [
+					".gitmodules",
+					"Cartfile",
+					"Cartfile.lock",
+					"Cartfile.private",
+					"Cartfile.resolved",
+					"Carthage.checkout/Nimble",
+					"Carthage.checkout/Quick",
+					"Carthage.checkout/xcconfigs",
+					"Carthage/Checkouts/Nimble",
+					"Carthage/Checkouts/Quick",
+					"Carthage/Checkouts/xcconfigs",
+					"Mantle.xcodeproj/project.pbxproj",
+					"Mantle.xcworkspace/contents.xcworkspacedata",
+				]
+				let expectedOldFilePaths = [
+					".gitmodules",
+					"Cartfile",
+					"Cartfile.lock",
+					"Cartfile.private",
+					"Cartfile.resolved",
+					"Carthage.checkout/Nimble",
+					"Carthage.checkout/Quick",
+					"Carthage.checkout/xcconfigs",
+					"Carthage/Checkouts/Nimble",
+					"Carthage/Checkouts/Quick",
+					"Carthage/Checkouts/xcconfigs",
+					"Mantle.xcodeproj/project.pbxproj",
+					"Mantle.xcworkspace/contents.xcworkspacedata",
+				]
+
+				let repo = Fixtures.mantleRepository
+				let branch = repo.localBranch(named: "master").value!
+				expect(repo.checkout(branch, strategy: CheckoutStrategy.None).error).to(beNil())
+
+				let head = repo.HEAD().value!
+				let commit = repo.object(head.oid).value! as! Commit
+				let deltas = repo.diff(for: commit).value!
+
+				var newFilePaths: [String] = []
+				for delta in deltas {
+					newFilePaths.append((delta.newFile?.path)!)
+				}
+				var oldFilePaths: [String] = []
+				for delta in deltas {
+					oldFilePaths.append((delta.oldFile?.path)!)
+				}
```

Comment:
These could use `map` instead of `for` loops

```swift
let oldFilePaths = deltas.map { $0.oldFile!.path }
```

Reply:
nice :+1: